src/include/platform/internal/GenericPlatformManagerImpl.cpp

```diff
@@ -298,7 +298,10 @@ void GenericPlatformManagerImpl<ImplClass>::HandleDeviceRebooted(intptr_t arg)
 
     if (generalDiagnosticsDelegate != nullptr)
     {
-        generalDiagnosticsDelegate->OnDeviceRebooted();
+        uint8_t bootReason;
+
+        ReturnOnFailure(GetDiagnosticDataProvider().GetBootReason(bootReason));
```

Comment:
I think the boot reason may not be available on each platform. How about just skipping this event rather than returning from the function?